src/sage/rings/function_field/drinfeld_modules/charzero_drinfeld_module.py

```diff
@@ -328,7 +352,13 @@ def coeff_log(k):
                 return self._compute_coefficient_log(v)
             else:
                 return self._base.zero()
-        return L(coeff_log, valuation=1)
+
+        if prec is Infinity:
+            L = LazyPowerSeriesRing(self._base, name)
+            return L(coeff_log, valuation=1)
+        else:
+            L = PowerSeriesRing(self._base, name, default_prec=prec)
+            return L([0] + [coeff_log(i) for i in range(1, prec)], prec=prec)
```

Comment:
```suggestion
        L = PowerSeriesRing(self._base, name, default_prec=prec)
        return L([0] + [coeff_log(i) for i in range(1, prec)], prec=prec)
```
And also in `coeff_log`.